Sources/SwiftProtobuf/ProtobufBinaryTypes.swift

```diff
@@ -789,14 +777,21 @@ public extension ProtobufBinaryMessageBase {
 
 public extension ProtobufMessage {
     static func decodeProtobufMapValue(decoder: inout ProtobufFieldDecoder, value: inout Self?) throws {
-        let handled = try decoder.decodeSingularMessageField(fieldType: Self.self, value: &value)
-        assert(handled)
+        try decoder.decodeSingularMessageField(fieldType: Self.self, value: &value)
+        assert(value != nil)
     }
 
     init(protobuf: Data) throws {
         try self.init(protobuf: protobuf, extensions: nil)
     }
 
+    init(protobufBuffer: UnsafeBufferPointer<UInt8>) throws {
+        try self.init(protobufBuffer: protobufBuffer, extensions: nil)
+    }
+}
+
+// For Proto3 messages, we need to preserve the unknown fields when we're done
```

Comment:
For ObjC, I we didn't add a different base class because most ObjC code that deals with generic messages would still use the common base between them for passing messages.  With the dynamic runtime, adding a second class would actually make things a bit more complex on the runtime size.  So just letting the methods return empty sets felt a little more ObjC-ish (just like messaging nil, etc.) and keep things simple.

There also was very little interesting in proto3 syntax (over proto2) when the work was initially being done.  The unknown support for mobile is actually pretty useful as it means a server can get feed extra data, clients to store it; and then when the users update their app, the data is there.  With proto3, developers have to write extra logic to go refetch things if they could now have extra info they care about.